### PR TITLE
addLink: support dynamic links added from behaviors

### PR DESCRIPTION
**Pull request by ikreymer**
autoscroll:
- detect if page url (pathname, not hash) changes when scrolling and abort scroll
- support link extraction via addLink(), if available
- add newly changed URL for future crawling with addLink()

general:
- allow behaviors to run in top replay frame (if running within wombat-based replay system) by checking _WB_replay_top as alternative to window.top

bump to 0.5.3

### CODE REVIEW

File: src/autoscroll.ts [hunk 1]
```diff
@@ -113,6 +117,13 @@ export class AutoScroll extends Behavior {
     let lastScrollHeight = self.document.scrollingElement.scrollHeight;
 
     while (this.canScrollMore()) {
+      if (document.location.pathname !== this.origPath) {
+        behaviorLog("Location Changed, stopping scroll: " +
+          `${document.location.pathname} != ${this.origPath}`, "info");
+        addLink(document.location.href);
```
Comment (tw4l) [inline src/autoscroll.ts:123]: If the location has already changed by this point, should we grab the URL prior to this?

File: src/index.ts [hunk 2]
```diff
@@ -136,9 +136,9 @@ export class BehaviorManager {
     if (this.mainBehavior) {
       this.behaviors.push(this.mainBehavior);
 
-      if (this.mainBehavior instanceof Behavior) {
+      // if (this.mainBehavior instanceof Behavior) {
 
-      }
+      // }
```
Comment (tw4l) [inline src/index.ts:141]: What's the point of keeping this commented out?

**Review by tw4l — COMMENTED** [review]
Left a few comments, otherwise looks good

File: src/autoscroll.ts (hunk 1 above)
Comment (ikreymer) [reply src/autoscroll.ts:123]: This is called in a loop while page is scrolling, so should be the quickest we detect the change due to scrolling. I suppose we can add a more generic behavior that detects this not tied to scrolling, but for now this should be fine.

File: src/index.ts (hunk 2 above)
Comment (ikreymer) [reply src/index.ts:141]: Hm, forget now, perhaps should remove!

**Pull request merged**